cmd/all-in-one/main.go

```diff
@@ -175,9 +176,16 @@ func startAgent(
 	baseFactory metrics.Factory,
 ) {
 	metricsFactory := baseFactory.Namespace("agent", nil)
-
-	if len(b.CollectorHostPorts) == 0 {
-		b.CollectorHostPorts = append(b.CollectorHostPorts, fmt.Sprintf("127.0.0.1:%d", cOpts.CollectorPort))
+	fmt.Println("all main")
+	fmt.Printf("%v", b.ReporterBuilders)
+	fmt.Println(len(b.ReporterBuilders))
+	for _, rb := range b.ReporterBuilders {
+		if tb, ok := rb.(agentRepTchan.Builder); ok {
+			fmt.Println("adding reporter\n\n")
+			if len(tb.CollectorHostPorts) == 0 {
+				tb.CollectorHostPorts = append(tb.CollectorHostPorts, fmt.Sprintf("127.0.0.1:%d", cOpts.CollectorPort))
+			}
+		}
```

Comment:
There's no other possible reporter builder yet, but you never know the future... Perhaps it would be better to add a default case where, to log a warning in in case the type isn't recognized.